azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VaultApiMockTest.java

```diff
@@ -0,0 +1,2845 @@
+      super.start();
+      try {
+         vaultUri = server.getUrl("").toURI();
+      } catch (URISyntaxException use) {
+      }
+   }
+
+   public void createVault() throws InterruptedException {
+      server.enqueue(jsonResponse("/vaultcreate.json").setResponseCode(200));
+      final VaultApi vaultApi = api.getVaultApi(resourceGroup);
+      Vault vault = vaultApi.createOrUpdateVault(vaultName, location, VaultProperties.builder()
+              .tenantId(this.tenantId)
+              .sku(SKU.create(location, "standard", null, "A"))
+              .accessPolicies(ImmutableList.of(VaultProperties.AccessPolicyEntry.create(null, this.identityObjId, this.tenantId,
+                      VaultProperties.Permissions.create(
+                              ImmutableList.of( // certificates
+                                      "Get",
+                                      "List",
+                                      "Update",
+                                      "Create",
+                                      "Import",
+                                      "Delete",
+                                      "ManageContacts",
+                                      "ManageIssuers",
+                                      "GetIssuers",
+                                      "ListIssuers",
+                                      "SetIssuers",
+                                      "DeleteIssuers",
+                                      "Purge",
+                                      "Recover"
+                              ),
+                              ImmutableList.of( // keys
+                                      "Get",
+                                      "List",
+                                      "Update",
+                                      "Create",
+                                      "Import",
+                                      "Delete",
+                                      "Recover",
+                                      "Backup",
+                                      "Restore",
+                                      "Purge"
+                              ),
+                              ImmutableList.of( // secrets
+                                      "Get",
+                                      "List",
+                                      "Set",
+                                      "Delete",
+                                      "Recover",
+                                      "Backup",
+                                      "Restore",
+                                      "Purge"
+                              ),
+                              ImmutableList.<String>of()
+                      ))))
+              .build(),
+              null);
+
+      String path = String.format(
+              "/subscriptions/%s/resourcegroups/%s/providers/Microsoft.KeyVault/vaults/%s?%s",
+              subscriptionId, resourceGroup, vaultName, apiVersion
+      );
+      assertSent(server, "PUT", path);
```

Comment:
I know this can be kind of a pain, but we need to assert the sent payload too for all requests that have a body. This *really* helps future maintenance of the different APIs.